src/compiler/checker.ts

```diff
@@ -14748,21 +14748,70 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
      * Return the signatures of the given kind in the given type. Creates synthetic union signatures when necessary and
      * maps primitive types and type parameters are to their apparent types.
      */
-    function getSignaturesOfType(type: Type, kind: SignatureKind): readonly Signature[] {
-        const result = getSignaturesOfStructuredType(getReducedApparentType(type), kind);
-        if (kind === SignatureKind.Call && !length(result) && type.flags & TypeFlags.Union) {
-            if ((type as UnionType).arrayFallbackSignatures) {
-                return (type as UnionType).arrayFallbackSignatures!;
-            }
-            // If the union is all different instantiations of a member of the global array type...
-            let memberName: __String;
-            if (everyType(type, t => !!t.symbol?.parent && isArrayOrTupleSymbol(t.symbol.parent) && (!memberName ? (memberName = t.symbol.escapedName, true) : memberName === t.symbol.escapedName))) {
-                // Transform the type from `(A[] | B[])["member"]` to `(A | B)[]["member"]` (since we pretend array is covariant anyway)
-                const arrayArg = mapType(type, t => getMappedType((isReadonlyArraySymbol(t.symbol.parent) ? globalReadonlyArrayType : globalArrayType).typeParameters![0], (t as AnonymousType).mapper!));
-                const arrayType = createArrayType(arrayArg, someType(type, t => isReadonlyArraySymbol(t.symbol.parent)));
-                return (type as UnionType).arrayFallbackSignatures = getSignaturesOfType(getTypeOfPropertyOfType(arrayType, memberName!)!, kind);
-            }
-            (type as UnionType).arrayFallbackSignatures = result;
+    function getSignaturesOfType(typeIn: Type, kind: SignatureKind): readonly Signature[] {
+        const reducedType = getReducedApparentType(typeIn);
+        function carveoutResult(): readonly Signature[] | undefined {
+            if (kind === SignatureKind.Call && reducedType.flags & TypeFlags.Union) {
+                // If the union is all different instantiations of a member of the global array type...

```

Comment:
This carveout was originally introduced in #53480.  Previously it was only used in the case where `getSignaturesOfStructuredType()` return a length zero result: `!length(result)`.

With new filter overload,  `getSignaturesOfStructuredType()` returns an invalid result of length 1.  Therefore, in order to piggyback onto this carveout to solve the problem, the condition would have to be expanded to inlcude at least length 1, at least for the case of method `filter`.

Considering the benefits of having the type behavior of array methods predicable and uniform, the length condition was dropped altogether, and the extension was not limited to just `filter`.